CONTRIBUTING.md: Add nixpkgs-merge-bot usage paragraph (#305553)
NixOS/nixpkgs: CONTRIBUTING.md

[pr-merge]: #how-to-merge-pull-requests

The *Nixpkgs committers* are people who have been given

> permission to merge. 

Reviewer: ```suggestion
permission to merge.
```

> To streamline autoupdates, leverage the nixpkgs-merge-bot by simply commenting @NixOS/nixpkgs-merge-bot merge if the package resides in pkgs-by-name and the commenter is among the package maintainers. The bot ensures that all ofborg checks, except for darwin, are successfully completed before merging the pull request. Should the checks still be underway, the bot patiently waits for ofborg to finish before attempting the merge again.

Reviewer 2: ```suggestion
To streamline autoupdates, leverage the nixpkgs-merge-bot by simply commenting `@NixOS/nixpkgs-merge-bot merge` if the package resides in `pkgs/by-name` and the commenter is among the package maintainers. The bot ensures that all ofborg checks, except for darwin, are successfully completed before merging the pull request. Should the checks still be underway, the bot patiently waits for ofBorg to finish before attempting the merge again.
```

The *Nixpkgs committers* are people who have been given
permission to merge.

> To streamline autoupdates, leverage the nixpkgs-merge-bot by simply commenting `@NixOS/nixpkgs-merge-bot merge` if the package resides in `pkgs/by-name` and the commenter is among the package maintainers. The bot ensures that all ofBorg checks, except for darwin, are successfully completed before merging the pull request. Should the checks still be underway, the bot patiently waits for ofBorg to finish before attempting the merge again.

Reviewer 3: Since there are discussions about the future roles of merge bot, I think it should be moire future-proof.

```suggestion
To streamline automated updates, leverage the nixpkgs-merge-bot by simply commenting `@NixOS/nixpkgs-merge-bot merge`. The bot will verify if the following conditions are met, refusing to merge otherwise:
- the commenter that issued the command should be among the package maintainers;
- the package should reside in `pkgs/by-name`.

Further, nixpkgs-merge-bot will ensure all ofBorg checks (except the Darwin-related ones) are successfully completed before merging the pull request. Should the checks still be underway, the bot patiently waits for ofBorg to finish before attempting the merge again.
```
